src/api/Controllers/CasingController.cs

```diff
@@ -67,13 +65,59 @@ public async Task<ActionResult<Casing>> GetByIdAsync(int id)
 
     /// <inheritdoc />
     [Authorize(Policy = PolicyNames.Viewer)]
-    public override Task<ActionResult<Casing>> CreateAsync(Casing entity)
-        => base.CreateAsync(entity);
+    public override async Task<ActionResult<Casing>> CreateAsync(Casing entity)
+    {
+        if (entity == null) return BadRequest();
+
+        try
+        {
+            if (!(entity.CasingElements?.Count > 0))
+            {
+                return Problem("At least one casing element must be defined.");
+            }
+
+            return await base.CreateAsync(entity).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            var message = "An error ocurred while creating the completion.";
+            Logger.LogError(ex, message);
+            return Problem(message);
+        }
+    }
 
     /// <inheritdoc />
     [Authorize(Policy = PolicyNames.Viewer)]
-    public override Task<ActionResult<Casing>> EditAsync(Casing entity)
-        => base.EditAsync(entity);
+    public override async Task<ActionResult<Casing>> EditAsync(Casing entity)
+    {
+        if (entity == null) return BadRequest();
+
+        try
+        {
+            if (entity.CasingElements.Count == 0)
+            {
+                return Problem("At least one casing element must be defined.");
+            }
+
+            var existingEntity = await Context.Casings.Include(c => c.CasingElements).SingleOrDefaultAsync(c => c.Id == entity.Id).ConfigureAwait(false);
+            if (existingEntity == null)
+            {
+                return NotFound();
+            }
+
+            Context.Entry(existingEntity).CurrentValues.SetValues(entity);
+            existingEntity.CasingElements = entity.CasingElements;
+
+            await Context.UpdateChangeInformationAndSaveChangesAsync(HttpContext).ConfigureAwait(false);
+            return Ok(entity);
+        }
+        catch (Exception ex)
+        {
+            var message = "An error ocurred while processing the casing";
```

Comment:
PP: Satzzeichen fehlt und Typo fixen
```suggestion
            var message = "An error occurred while updating the casing.";
```